File: pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
app.controller("brandController", function ($scope, $controller, brandService) {

    $controller("baseController", {$scope: $scope})

    $scope.findAll = function () {
        brandService.findAll().success(function (date) {
            $scope.list = date
        })
    }


    $scope.findPage = function (pageNum, pageSize) {
        brandService.findPage(pageNum, pageSize).success(function (date) {
            $scope.paginationConf.totalItems = date.total
            $scope.list = date.rows;//当前页查询结果集
        })
    }

    $scope.save = function () {
        var method = null;

        if ($scope.entity.id != null) {
            method = brandService.update($scope.entity)
        } else {
            method = brandService.add($scope.entity)
        }

        method.success(function (date) {
            if (date.success) {
                $scope.reloadList()
            } else {
                alert(date.message)
            }
        })
    }

    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (date) {
            $scope.entity = date
        })
    }

    $scope.dele = function () {
        brandService.dele($scope.selectIds).success(function () {
            $scope.reloadList()
        })
    }

})

File: pinyougou_manager_web/src/main/java/com/pinyougou/controller/BrandController.java
package com.pinyougou.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        List<TbBrand> brands = brandService.findAll();
        return brands;
    }

    @RequestMapping("/findPage")
    public PageResult findPage(Integer pageNum, Integer pageSize) {
        return brandService.findPage(pageNum, pageSize);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand) {
        try {
            brandService.add(tbBrand);
            return new Result(true, "ok");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "失败！！！");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id) {
        return brandService.findOne(id);
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand) {
        try {
            brandService.update(tbBrand);
            return new Result(true, "ok");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "失败！！！");
        }
    }

    @RequestMapping("/delee")
    public Result delete(Long[] ids) {
        System.out.println(ids);
        try {
            brandService.dele(ids);
            return new Result(true, "ok");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "失败！！！");
        }
    }

    @RequestMapping("/selectBrandList")
    public List<Map> tbBrandMapper() {
        return brandService.SelectBrandList();
    }

}